### PR TITLE
#5590 Introduce enhanced Server Author fetch platforms to include all the registered services. 

### PR DESCRIPTION
**Pull request by davidradl**
# Description
Amend the server author to return a fuller list of platforms.

Fixes # (issue) #5590

# How Has This Been Tested?

React UI

# Any additional notes for reviewers?
The issue talks of wider considerations in this area that may result in more prs for the issue

In addition to Server author changes:
 
- I have changed the application.properties to remove the default of Debug from the logging. This was confusing me as I had assumed that by default we would have no debug logging - so I assumed the entries were errors.

- I have amended the PlatformServers to add 2 new APIs.

- Dino has been amended to call the new Platform services APIs to get the integration and engine registered services as well as the other registered services. 

- server author has been introduced to the Gradle build as it was missing.

### CODE REVIEW

File: open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/MetadataAccessPointConfigurationClient.java [hunk 1]
```diff
@@ -71,13 +71,20 @@ public MetadataAccessPointConfigurationClient(String adminUserId,
...
+     * This call is deprecated, as the MetadataAccessPointConfigurationClient client is for configuring a particular
+     * server, not interrogating platform information.
+     *
+     * Please use the call getRegisteredAccessServices in the platform configuration client
+     * OMAGServerPlatformConfigurationClient.
      *
      * @return list of access service descriptions
      * @throws OMAGNotAuthorizedException the supplied userId is not authorized to issue this command.
      * @throws OMAGInvalidParameterException invalid parameter.
      * @throws OMAGConfigurationErrorException unusual state in the admin server.
```
Comment (mandy-chessell) [inline open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/MetadataAccessPointConfigurationClient.java:84]: This call is needed - do not deprecate

File: open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/OMAGServerPlatformConfigurationClient.java [hunk 2]
```diff
@@ -278,4 +281,30 @@ public Connection getPlatformSecurityConnection() throws OMAGNotAuthorizedExcept
...
+
+    /**
+     * Return the list of access services for this platform.
+     *
+     * @return list of access service descriptions registered on this platform
+     * @throws OMAGNotAuthorizedException the supplied userId is not authorized to issue this command.
+     * @throws OMAGInvalidParameterException invalid parameter.
+     * @throws OMAGConfigurationErrorException unusual state in the admin server.
+     */
+    public List<RegisteredOMAGService> getRegisteredAccessServices() throws OMAGNotAuthorizedException,
```
Comment (mandy-chessell) [inline open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/OMAGServerPlatformConfigurationClient.java]: I do not see the value of adding this call here -if it is not accompanied by equivalent calls for the other types of registered services - engine, view, integration

File: open-metadata-implementation/view-services/server-author-view/server-author-view-server/src/main/java/org/odpi/openmetadata/viewservices/serverauthor/handlers/ServerAuthorViewHandler.java [hunk 3]
```diff
@@ -185,6 +186,10 @@ public ServerAuthorViewHandler(String localServerUserId, String metadataServerUR
                         // do not have a description of the server yet.
                         platform.addStoredServer(storedServer);
                     }
+                    //`/open-metadata/platform-services/users/${userId}/server-platform/registered-services/access-services`;
+                    MetadataAccessPointConfigurationClient metadataAccessPointConfigurationClient = new MetadataAccessPointConfigurationClient(userId, "",resourceEndpoint.getResourceRootURL());
```
Comment (mandy-chessell) [inline open-metadata-implementation/view-services/server-author-view/server-author-view-server/src/main/java/org/odpi/openmetadata/viewservices/serverauthor/handlers/ServerAuthorViewHandler.java]: Strange to be only retrieving the access services.  Seems to be missing requests for retrieving other registered services needed for configuring other types of OMAG Server?

**Review by mandy-chessell — CHANGES_REQUESTED** [review]

File: open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/MetadataAccessPointConfigurationClient.java (hunk 1 above)
Comment (davidradl) [reply open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/MetadataAccessPointConfigurationClient.java:84]: OK - I am unsure what to put as the serverName in the constructor of this class. As the call is a platform orientated call without an associated server.
It ends up calling `AdminServicesRESTClient(String serverName,
                            String serverPlatformURLRoot) throws OMAGInvalidParameterException
    {
        final String  methodName = "RESTClient(no authentication)";

        this.serverPlatformURLRoot = serverPlatformURLRoot;

        RESTClientFactory factory = new RESTClientFactory(serverName, serverPlatformURLRoot);

        try
        {
            this.clientConnector = factory.getClientConnector();
        }
        catch (Exception error)
        {
            throw new OMAGInvalidParameterException(OMAGAdminErrorCode.NULL_LOCAL_SERVER_NAME.getMessageDefinition(serverName, error.getMessage()),
                                                    this.getClass().getName(),
                                                    methodName,
                                                    error);
        }
    }`
 The Exception and call seems to want a valid serverName.  Am I missing something?

File: open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/OMAGServerPlatformConfigurationClient.java (hunk 2 above)
Comment (davidradl) [reply open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/OMAGServerPlatformConfigurationClient.java]: I have removed it now - I found the PlatformServices Client.

File: open-metadata-implementation/view-services/server-author-view/server-author-view-server/src/main/java/org/odpi/openmetadata/viewservices/serverauthor/handlers/ServerAuthorViewHandler.java (hunk 3 above)
Comment (davidradl) [reply open-metadata-implementation/view-services/server-author-view/server-author-view-server/src/main/java/org/odpi/openmetadata/viewservices/serverauthor/handlers/ServerAuthorViewHandler.java]: Yes I agree - I started duplicating the capability of the existing code, I realised the PlatformServices Client that Dino uses was a better place to get this information

File: open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/MetadataAccessPointConfigurationClient.java (hunk 1 above)
Comment (mandy-chessell) [reply open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/MetadataAccessPointConfigurationClient.java:84]: It is behaving correctly for the usecase it is designed for - ie configuring a server.  In this case the server name is the server being configured.  Requesting the registered access services is just one of the sequence of calls used to create a metadata access point and/or metadata server.

Use the platform services client if not configuring a specific server.

File: settings.gradle [hunk 4]
```diff
@@ -256,6 +256,10 @@ include(':open-metadata-implementation:view-services:dino-view:dino-view-api')
 include(':open-metadata-implementation:view-services:dino-view:dino-view-server')
 include(':open-metadata-implementation:view-services:dino-view:dino-view-spring')
 include(':open-metadata-implementation:view-services:dino-view')
+include(':open-metadata-implementation:view-services:server-author-view:server-author-view-api')
+include(':open-metadata-implementation:view-services:server-author-view:server-author-view-server')
+include(':open-metadata-implementation:view-services:server-author-view:server-author-view-spring')
+include(':open-metadata-implementation:view-services:server-author-view')
```
Comment (planetf1) [inline settings.gradle:262]: Great so far, but we also need project definitions ie similar to these we use for dino:
```
project(':open-metadata-implementation:view-services:dino-view:dino-view-server').projectDir = file('open-metadata-implementation/     view-services/dino-view/dino-view-server')
```
You'll need to also add those to settings.gradle for the above projects.

We shouldn't need the top level directory if empty (server-author-view) but we still have them for other projects, so ok to leave for consistency (I'll clean this area up later when I get back to more gradle work)

**Review by planetf1 — CHANGES_REQUESTED** [review]

File: open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/MetadataAccessPointConfigurationClient.java (hunk 1 above)
Comment (davidradl) [reply open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/MetadataAccessPointConfigurationClient.java:84]: ok will do

Comment (davidradl) [reply open-metadata-implementation/admin-services/admin-services-client/src/main/java/org/odpi/openmetadata/adminservices/client/MetadataAccessPointConfigurationClient.java]: done

File: settings.gradle (hunk 4 above)
Comment (davidradl) [reply settings.gradle:262]: ok done

**Pull request merged**